### PR TITLE
[CodeGen][Tablegen] Fix uninitialized var and shift overflow.

### PR DESCRIPTION
**Pull request by mahesh-attarde**
Fix uninitialized var and shift overflow.

### CODE REVIEW

File: llvm/utils/TableGen/DecoderEmitter.cpp [hunk 1]
```diff
@@ -935,7 +935,7 @@ void DecoderEmitter::emitTable(formatted_raw_ostream &OS, DecoderTable &Table,
       unsigned Shift = 0;
       do {
         OS << ", " << (unsigned)*I;
-        Value += (*I & 0x7f) << Shift;
+        Value += ((uint64_t)(*I & 0x7f)) << Shift;
```
Comment (phoebewang) [inline llvm/utils/TableGen/DecoderEmitter.cpp:938]: Not necessary. We can confirm (*I & 0x7f) is safe to do shift.

Comment (KanRobert) [reply llvm/utils/TableGen/DecoderEmitter.cpp:938]: +1

Comment (mahesh-attarde) [reply llvm/utils/TableGen/DecoderEmitter.cpp:938]: ok

Comment (phoebewang) [reply llvm/utils/TableGen/DecoderEmitter.cpp:938]: Sorry, I think the (uint64_t) in needed, because `Value` is uint64_t and `Shift` may larger than 32.

Comment (mahesh-attarde) [reply llvm/utils/TableGen/DecoderEmitter.cpp:938]: i see. there are no pending issues here, can you approve & merge request?

**Review by phoebewang — APPROVED** [review]
LGTM.

**Pull request merged**